### PR TITLE
Add liggght patched for newer compiler

### PR DESCRIPTION
**Pull request by SofiaXu**
Add C++ 17 support
Add Clang and Oneapi support

### CODE REVIEW

File: var/spack/repos/builtin/packages/liggghts/package.py
Comment (alecbcs) [inline var/spack/repos/builtin/packages/liggghts/package.py]:
```suggestion
    maintainers("SofiaXu")
    
    version("3.8.0", sha256="9cb2e6596f584463ac2f80e3ff7b9588b7e3638c44324635b6329df87b90ab03")
```

@SofiaXu thanks for the fix! Just a quick suggestion, often times maintainers are listed before versions to make it a bit easier to find the maintainer in packages with lots of versions.

File: var/spack/repos/builtin/packages/liggghts/package.py
Comment (alecbcs) [inline var/spack/repos/builtin/packages/liggghts/package.py:68]: @SofiaXu is this a recent change in the makefile? If so we should look at scoping these changes to only the most recent version.

Comment (alecbcs) [reply var/spack/repos/builtin/packages/liggghts/package.py:68]: ping @SofiaXu

Comment (SofiaXu) [reply var/spack/repos/builtin/packages/liggghts/package.py:68]: Sorry for waitting so long. It is usr from v3.6.0 when the file(src/MAKE/Makefile.user_default) was created. You can check the file on https://github.com/CFDEMproject/LIGGGHTS-PUBLIC/blob/18a96adf8688e0d46423c7d812001a8d9e0a9af0/src/MAKE/Makefile.user#L48

**Review by alecbcs — DISMISSED** [review]
Looks good to me. @SofiaXu thanks for working on this!

Looks like there's a merge conflict. Happy to merge once that's fixed however.

**Comment by SofiaXu** [comment]
> Looks good to me. @SofiaXu thanks for working on this!
> 
> Looks like there's a merge conflict. Happy to merge once that's fixed however.

@alecbcs Fixed. But I have some problem when build vtk 8.2.0 or 8.1.2 in my Ubuntu 22.04 with gcc 11. I think it may pass in ci since they just changed a name of zlib.

**Review by alecbcs — APPROVED** [review]
Looks good to me.

**Pull request merged**
